src/unwrap_rng.rs

```diff
@@ -0,0 +1,28 @@
+#![allow(clippy::unwrap_used)]
+use rand_core::{CryptoRng, RngCore, TryCryptoRng, TryRngCore};
+
+/// Wrapper around [`TryRngCore`] implementation which implements [`RngCore`]
+/// by panicking on potential errors.
+///
+/// Similar to [`rand_core::UnwrapErr`] but works with inner references.
+#[derive(Debug, Eq, PartialEq, Hash)]
+pub(crate) struct UnwrapRng<'r, R: TryRngCore>(pub &'r mut R);
```

Comment:
This is only here because [`signature::RandomizedDigestSigner`](https://docs.rs/signature/2.3.0-pre.5/signature/trait.RandomizedDigestSigner.html#tymethod.try_sign_digest_with_rng) forces us to accept a `TryCryptoRng` but the inner rsa implementation for pss and pkcs1.5 only work with `CryptoRng`.

Reply:
Yeah, and maybe we should upstream that?

I have to bring it over in [ecdsa too](
https://github.com/RustCrypto/signatures/pull/901/files#diff-a5d3287dfb8311c23907e9e235831d4c651c1e73e69a3153c51b9db1e292f9ef)

Reply:
Let's see upstream's opinion on that: https://github.com/rust-random/rand/pull/1589

Reply:
Maybe it's better to switch the internal implementation to `TryCryptoRng` instead?

Reply:
we might need to tweak a bit the [RandomizedPrehashSigner](https://docs.rs/signature/2.3.0-pre.5/signature/hazmat/trait.RandomizedPrehashSigner.html) to accept `TryCryptoRng` then

Reply:
Sounds good. It should be a simple enough tweak considering hat the methods return `Result`.

Reply:
I'm not sure I can change the inner implementation of RSA to accept a `TryCryptoRng`. This calls into `crypto-bigint` and this would require a larger redesign than I'm comfortable with.